File: specificIDEprojectz/blueJ/Numbers/Prime.java
/* Accept a number and check whether it is a prime 
 * number or not
 */
public class Prime
{
public static void main(int n)
{
int i;
for(i=2;i<n;i++)
{
if(n%i==0)
{
break;
}
}
if(i==n)
System.out.println(n+" is a prime number");
else
System.out.println(n+" is not a prime number");
}
}

File: sourceCodez/C/prime.c
#include<stdio.h>

// Returns the number of factors of n.
int numberOfFactors(int n) {

	int count = 0;

	for (int i = 1; i <= n; i++) {

		if (n % i == 0) {
			count++;
		}
	}

	return count;
}

// Will return 1 if n is prime.
// Will return 0 if n is not prime.
int isPrime(int n) {

	if (numberOfFactors(n) == 2) {
		return 1;
	}

	return 0;
}

int main() {

	int n;
	printf("number: ");
	scanf("%d", &n);

	if (isPrime(n)) {
		printf("prime number\n");
	} else {
		printf("not a prime number\n");
	}

	return 0;
}